Tests/ComposableNavigationTests/Helper/XCTestCase+RetainCycle.swift

```diff
@@ -1,15 +1,52 @@
 import Foundation
 import XCTest
+import ComposableArchitecture
 
 public extension XCTestCase {
-	func assertNil(_ subject: AnyObject?, after: @escaping () -> Void, file: StaticString = #file, line: UInt = #line) {
+	func assertNil(
+		_ subject: AnyObject?,
+		after: @escaping () -> Void,
+		file: StaticString = #filePath,
+		line: UInt = #line
+	) {
 		guard let value = subject else {
 			return XCTFail("Argument must not be nil", file: file, line: line)
 		}
-
-		addTeardownBlock { [weak value] in
-			after()
-			XCTAssert(value == nil, "Expected subject to be nil after test! Retain cycle?", file: file, line: line)
+		let afterContainer = LockContainer(after)
+		let weakContainer = WeakLockContainer(value)
+		addTeardownBlock {
+			afterContainer.value()
+			XCTAssert(weakContainer.isValueNil, "Expected subject to be nil after test! Retain cycle?", file: file, line: line)
+		}
+	}
+	
+	private class LockContainer<Value>: @unchecked Sendable {
```

Comment:
In the assertNil signature, after is defined as a closure so what do you think if we change the LockContainer to something like :
```
private class LockClosureContainer {
        private var closure: () -> Void
        private let lock = NSRecursiveLock()

        init(_ closure: @escaping () -> Void) {
            self.closure = closure
        }

        func perform() {
            lock.lock()
            defer { lock.unlock() }
            closure()
        }
    }
```
and use afterContainer.perform() inside addTeardownBlock?